Mathlib/Data/Fintype/Basic.lean

```diff
@@ -936,6 +936,8 @@ theorem Fintype.coe_image_univ [Fintype α] [DecidableEq β] {f : α → β} :
 instance List.Subtype.fintype [DecidableEq α] (l : List α) : Fintype { x // x ∈ l } :=
   Fintype.ofList l.attach l.mem_attach
 
+instance List.setOf_fintype (l : List α) [h : Fintype { x // x ∈ l }] : Fintype { x | x ∈ l } := h
```

Comment:
This instance worries me a little, and might be worth asking Zulip about (apologies if I missed a previous conversation).

Reply:
[It was discussed slightly here](https://github.com/leanprover-community/mathlib4/pull/20488#discussion_r1910367396).